web/src/components/User/SignIn/SignInForm.js

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react'
+
+import Button from '@material-ui/core/Button'
+import CircularProgress from '@material-ui/core/CircularProgress'
+import FormControl from '@material-ui/core/FormControl'
+import IconButton from '@material-ui/core/IconButton'
+import Input from '@material-ui/core/Input'
+import InputAdornment from '@material-ui/core/InputAdornment'
+import InputLabel from '@material-ui/core/InputLabel'
+import Typography from '@material-ui/core/Typography'
+import Visibility from '@material-ui/icons/Visibility'
+import VisibilityOff from '@material-ui/icons/VisibilityOff'
+import { authUser } from '../../../actions/authActions'
+import { axiosInstance } from '../../../utils/axiosConfig'
+import { connect } from 'react-redux'
+import { makeStyles } from '@material-ui/core/styles'
+
+const useStyles = makeStyles({
+  form: {
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+    width: '350px',
+  },
+  fields: {
+    display: 'flex',
+    flexDirection: 'column',
+    width: '325px',
+  },
+  field: {
+    margin: '8px 0',
+  },
+  submitButton: {
+    marginTop: '24px',
+    width: '200px',
+  },
+})
+
+function SignInForm(props) {
```

Comment:
are we gonna have a rule that is H00Ks only

Reply:
Well there's a time and place to use each, hooks don't completely replace class based components. I'd start everything off as a hook-based component, and only switch to class if you find that you need it

Reply:
idk i had a chat with a buddy and he was talking about how like hooks are the future. Do you have an example were you would need a class? 

I guess other than if you need weird uncommon lifecycle methods? 

https://reactjs.org/docs/hooks-faq.html#should-i-use-hooks-classes-or-a-mix-of-both 

**Do Hooks cover all use cases for classes?**
`Our goal is for Hooks to cover all use cases for classes as soon as possible. There are no Hook equivalents to the uncommon getSnapshotBeforeUpdate, getDerivedStateFromError and componentDidCatch lifecycles yet, but we plan to add them soon.`

Reply:
https://stackoverflow.com/questions/54912470/will-react-hooks-replace-classes

Reply:
@Teddarific bump legitly curious if you had any thoughts on when you'd use a class-based component over a hook unless u were doing some really funky, uncommon thing?